vulnfeeds/cmd/combine-to-osv/main.go

```diff
@@ -241,3 +234,41 @@ func loadAllCVEs(cvePath string) map[cves.CVEID]cves.Vulnerability {
 	}
 	return result
 }
+
+// addReference adds the related security tracker URL to a given vulnerability's references
+func addReference(cveId string, ecosystem string, convertedCve *vulns.Vulnerability) {
+	var securityReference vulns.Reference
+	securityReference.Type = "ADVISORY"
+	if ecosystem == "Alpine" {
+		securityReference.URL = fmt.Sprintf("https://security.alpinelinux.org/vuln/%s", cveId)
+	} else if ecosystem == "Debian" {
+		securityReference.URL = fmt.Sprintf("https://security-tracker.debian.org/tracker/%s", cveId)
+	}
+
+	if securityReference.URL != "" {
+		// Validate URL
+		req, err := http.NewRequest("HEAD", securityReference.URL, nil)
+		if err != nil {
+			Logger.Warnf("Unable to validate the URL '%s'", securityReference.URL)
+			return
+		}
+
+		// security.alpinelinux.org responds with text/html content.
+		req.Header.Set("Accept", "text/html")
+
+		// Send the request
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			Logger.Warnf("Unable to validate the URL '%s'", securityReference.URL)
+			return
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode == http.StatusOK {
```

Comment:
See https://github.com/google/osv.dev/blob/260a49adcac4d7d06d75b2f53c0915225e15518e/vulnfeeds/cves/versions.go#L606 for how to do retries if you want to get all fancy and be fault tolerant of intermittent 500 errors...

Reply:
Thanks! Updated.

Reply:
Give you're already importing cves, can you avoid embedding the code from `cves.ValidateAndCanonicalizeLink()` and use it directly? I think the function is generic enough to meet your needs here.

Reply:
The only issue with `cves.ValidateAndCanonicalizeLink()` is that it internally makes an `http.Head()` request, which fails (with a `406` response code) for Alpine Linux URLs like `https://security.alpinelinux.org/vuln/CVE-2024-32760`. These URLs require an `(Accept: text/html)` header to be sent. I will need to modify the existing validation method to use `http.NewRequest()` and include the necessary header.

```
curl -I https://security.alpinelinux.org/vuln/CVE-2024-32760
HTTP/2 406 
content-type: text/html; charset=utf-8

curl -I --header "Accept: */*" https://security.alpinelinux.org/vuln/CVE-2024-32760
HTTP/2 406 
content-type: text/html; charset=utf-8

curl -I --header "Accept: text/html" https://security.alpinelinux.org/vuln/CVE-2024-32760
HTTP/2 200 
content-type: text/html; charset=utf-8
```

Reply:
That seems safe and backwards compatible to do (and unfortunately annoying)